File: client/models/Prize.ts
import { DonorForReadPrizesDTO } from './Donor';
import { ReadWinnerInPrizeDTO } from './Winner'

export interface CategoryDTOWithId {
  id: number;
  name: string;
}

export interface CreatePrizeDTO {
  name: string;
  description?: string;
  donorId: number;
  categoryIds: number[];
  imagePath?: string;
  qty: number;
}

export interface ReadPrizeDTO {
  id: number;
  name?: string;
  description?: string;
  donor?: DonorForReadPrizesDTO;
  categories?: CategoryDTOWithId[];
  imagePath?: string;
  qty: number;
  numOfTickets: number;
  winners: ReadWinnerInPrizeDTO[]

}

export interface UpdatePrizeDTO {
  id: number;
  name?: string;
  description?: string;
  donor?: DonorForReadPrizesDTO;
  categoryIds?: number[];
  imagePath?: string;
  qty: number;

}

export interface ReadSimplePrizeDTO {
  id: number;
  name: string;
  categories?: string[];
  imagePath?: string;
  winners: ReadWinnerInPrizeDTO[]
}

export interface PrizeForWinnerDTO {
  id: number;
  name: string;
  CategoriesNames?: string[];
  imagePath?: string;
}

File: server/Models/Prize.cs
using System.ComponentModel.DataAnnotations;

namespace ChineseAuctionAPI.Models
{
    public class Prize
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "Name is required"), MaxLength(100)]
        public string Name { get; set; }

        [Required(ErrorMessage = "Description is required"), MaxLength(500)]
        public string Description { get; set; }

        [Required(ErrorMessage = "Donor ID is required")]
        public int DonorId { get; set; }
        public Donor Donor { get; set; }
        
        public List<Category> Categories {get;set;}=new List<Category>();

        public string ImagePath {  get; set; }

        [Required(ErrorMessage = "Quantity is required"), Range(1, 50)]
        public int Qty { get; set; }

        public List<Order> Orders { get; set; }

        public List<Ticket> Tickets { get; set; }


        public List<Winner> Winners { get; set; }

         


    }
}
